ui/raidboss/data/06-ew/raid/p12s.ts

```diff
@@ -114,6 +132,10 @@ export interface Data extends RaidbossData {
   expectedFirstHeadmarker?: string;
   isDoorBoss: boolean;
   phase?: 'superchain1' | 'palladion' | 'superchain2a' | 'superchain2b';
+  engravementCounter: number;
+  engravement1Towers: string[];
+  engravement3TowerType?: string;
```

Comment:
```suggestion
  engravement3TowerType?: 'lightTower' | 'darkTower';
```

Ideally this sort of thing can be more strongly typed when there's only a few values.

You can do this more explicit sort of thing below as well:
```diff
-data.engravement3TowerType = engravementLabelMap[matches.effectId];
+if (matches.effectId === engravementIdMap.lightTower)
+  data.engravement3TowerType = 'lightTower';
+if (matches.effectId === engravementIdMap.darkTower)
+  data.engravement3TowerType = 'darkTower';
```

Reply:
> You can do this more explicit sort of thing below as well:

Since the netRegex is restricted to just the lightTower and darkTower effects (e.g. `engravementTowerIds`), are you generally okay with something more shorthand like:
`data.engravement3TowerType = matches.id === engravementIdMap.lightTower ? 'lightTower' : 'darkTower';`

Or do you prefer separate explicit conditionals?

Reply:
Yeah, that's fine too!